zebra-utils/src/bin/search-issue-refs/main.rs

```diff
@@ -0,0 +1,223 @@
+                let matching_text = potential_issue_ref.as_str();
+                let id = matching_text[matching_text.len() - 4..].to_string();
+
+                PossibleIssueRef {
+                    file_path: file_path.clone(),
+                    line_number: line_idx + 1,
+                    column: potential_issue_ref.start() + 1,
+                    id,
+                }
+            }))
+        }
+    }
+
+    let num_possible_issue_refs = possible_issue_refs.len();
+
+    println!(
+        "\nFound {num_possible_issue_refs} possible issue refs, checking Github issue statuses..\n"
+    );
+
+    // check if issues are closed on Github
+
+    let github_token = match env::vars().find(|(key, _)| key == GITHUB_TOKEN_ENV_KEY) {
+        Some((_, github_token)) => github_token,
+        _ => {
+            println!(
+                "Can't find {GITHUB_TOKEN_ENV_KEY} in env vars, printing all found possible issue refs"
+            );
+
+            for issue in possible_issue_refs {
+                println!("{issue:?}");
+            }
+
+            return Ok(());
+        }
+    };
+
+    let mut headers = HeaderMap::new();
+    let mut auth_value = HeaderValue::from_str(&format!("Bearer {github_token}"))?;
+    let accept_value = HeaderValue::from_static("application/vnd.github+json");
+    let github_api_version_value = HeaderValue::from_static("2022-11-28");
+    let user_agent_value = HeaderValue::from_static("search-issue-refs");
+
+    auth_value.set_sensitive(true);
+
+    headers.insert(header::AUTHORIZATION, auth_value);
+    headers.insert(header::ACCEPT, accept_value);
+    headers.insert("X-GitHub-Api-Version", github_api_version_value);
+    headers.insert(header::USER_AGENT, user_agent_value);
+
+    let client = ClientBuilder::new().default_headers(headers).build()?;
+
+    let mut github_api_requests = JoinSet::new();
+    let mut num_possible_issue_refs = 0;
+
+    for possible_issue_ref in possible_issue_refs {
+        let request = client
+            .get(github_issue_api_url(&possible_issue_ref.id))
+            .send();
+        github_api_requests.spawn(async move { (request.await, possible_issue_ref) });
+    }
+
+    while let Some(res) = github_api_requests.join_next().await {
+        let Ok((
+            Ok(res),
+            PossibleIssueRef {
+                file_path,
+                line_number,
+                column,
+                id,
+            },
+        )) = res else {
+            println!("warning: no response from github api");
+            continue;
+        };
+
+        let Ok(text) = res.text().await else {
+            println!("warning: failed to get text from response");
+            continue;
+        };
+
+        let Ok(json): Result<serde_json::Value, _> = serde_json::from_str(&text) else {
+            println!("warning: failed to get serde_json::Value from response");
+            continue;
+        };
```

Comment:
###### Optional

It might be worth printing out for which `possible_issue_ref` this happened.

Reply:
Added in [`b6439dd`](https://github.com/ZcashFoundation/zebra/pull/6347/commits/b6439ddf51fefb6c0b691cc95da8bec9965f3d3d).